### ISSUE TITLE
NULL::text query for many states with `near` method

### ISSUE THREAD

**Issue by danielpclark** [alexreisner/geocoder#1002]
Gem Version: 1.3.1

 I don't know why the `near` method isn't working most of the time.  Running this test I get this output:
```ruby
Address.near("New York, NY").to_sql
# => "SELECT addresses.*, NULL::text AS distance, NULL::text AS bearing FROM \"addresses\"  WHERE (false)"

# For each of the 2 letter states do the following
US::STATES.each do |state|
  if Address.near("New York, #{state}").to_sql == \
    "SELECT addresses.*, NULL::text AS distance, NULL::text AS bearing FROM \"addresses\"  WHERE (false)"
    @null_count = @null_count.to_i + 1
  end
end
puts @null_count
# => 38
```

More than half of the queries are failing in this way.

**Comment by danielpclark** [comment]
In case it helps I have two Address records in the DB during this test.  Both for New York, NY.

**Comment by danielpclark** [comment]
I switched from Yandex to GeoCod.io and I still get the errors but for different states.

```ruby
US::STATES.each do |state|
  if Address.near("New York, #{state}").to_sql == \
    "SELECT addresses.*, NULL::text AS distance, NULL::text AS bearing FROM \"addresses\"  WHERE (false)"
    @null_count = @null_count.to_i + 1
  end
end
puts @null_count
# => 32
```

With Yandex "New York, NY" would consistently fail and "New York, VA" would work.  Now with GeoCod.io It is reversed.  VA brings up a NULL query and NY works.

```ruby
Address.near("New York, NY").to_sql
# => "SELECT addresses.*, 3958.755864232 * 2 * ASIN(SQRT(POWER(SIN((40.68863 - addresses.latitude) * PI() / 180 / 2), 2) + COS(40.68863 * PI() / 180) * COS(addresses.latitude * PI() / 180) * POWER(SIN((-74.018244 - addresses.longitude) * PI() / 180 / 2), 2))) AS distance, MOD(CAST((ATAN2( ((addresses.longitude - -74.018244) / 57.2957795), ((addresses.latitude - 40.68863) / 57.2957795)) * 57.2957795) + 360 AS decimal), 360) AS bearing FROM \"addresses\"  WHERE (addresses.latitude BETWEEN 40.399166433778305 AND 40.9780935662217 AND addresses.longitude BETWEEN -74.39998923744648 AND -73.63649876255352 AND (3958.755864232 * 2 * ASIN(SQRT(POWER(SIN((40.68863 - addresses.latitude) * PI() / 180 / 2), 2) + COS(40.68863 * PI() / 180) * COS(addresses.latitude * PI() / 180) * POWER(SIN((-74.018244 - addresses.longitude) * PI() / 180 / 2), 2)))) BETWEEN 0.0 AND 20)  ORDER BY distance ASC" 
Address.near("New York, VA").to_sql
# => "SELECT addresses.*, NULL::text AS distance, NULL::text AS bearing FROM \"addresses\"  WHERE (false)"
```

**Comment by alexreisner** [comment]
It looks like what's happening is that the city you're using as the center point of your search radius is not being found. Make sure the first argument to `near` can be geocoded successfully before you look for objects near it.

**Comment by danielpclark** [comment]
New York, NY is a valid location and it failed with Yandex.  Is there a built-in way to have the query raise an InvalidLocationError rather than quietly process a NULL query?

**Comment by alexreisner** [comment]
Without an API key [Yandex is limited to certain regions in Asia](https://github.com/alexreisner/geocoder#yandex-yandex).

There is no built-in way to raise an exception for a query returning zero results, but you could check for an empty response and raise your own exception or do something else.

**Comment by danielpclark** [comment]
I've implemented the `:near!` scope method in #1003 .  The test is there in the comment with results.  

Would you be okay with me adding the test and two additional gems to the Gemfile?  With my test script you'd be able to test all the other scope methods as well.  Although we might have to do something in case the test namespace would conflict with existing tests.  I may be able to do it.  Let me know.
